samcli/lib/build/app_builder.py

```diff
@@ -513,6 +513,11 @@ def _build_layer(
             manifest_path = self._manifest_path_override or os.path.join(manifest_context_path, config.manifest_name)
 
             # By default prefer to build in-process for speed
+            scratch_dir_path = (
```

Comment:
Why is this a change that needs to be made for terraform specific projects?

Reply:
this is not required only for terraform, but the scratch_dir is different in container, and if there is any runtime that depend on this value it will not work on containers.

Reply:
the question I ask is then should this be a change we need to make in `develop` branch as well?. Integration tests should have been failing if that is the case, so I'm thinking thats probably not the case.

Reply:
I think Terraform is the first thing that depend on the scratch directory on the logic.